### PR TITLE
fix: Proxy PATCH method with content

### PR DESCRIPTION
**Pull request by dluc**
# Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] The code follows the code style and conventions of this project.
- [ ] The change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

# Description of the change
The application was not including the payload when proxying requests using the PATCH method.

As a result PCS UI could not ack/close Alarms.

Extras:
* refactor the configuration so that all the settings can be optionally managed via environment variables. The remote hostname env var is the only mandatory env var, while the other are optional and fall back to a hard coded default compatible with IoT PCS scenarios.

---
This change is [Reviewable](https://reviewable.io/reviews/azure/reverse-proxy-dotnet/14)

### CODE REVIEW

File: ProxyAgent/ProxyMiddleware.cs [hunk 1]
```diff
@@ -31,6 +31,7 @@ public class ProxyMiddleware
             try
             {
                 await this.proxy.ProcessAsync(this.hostname, context.Request, context.Response);
+                this.log.Debug("--------------------------------------------------------------------------------", () => { });
```
Comment (jillcary) [inline ProxyAgent/ProxyMiddleware.cs]: What are the dashes for?

File: ProxyAgent/Runtime/Config.cs [hunk 2]
```diff
@@ -40,33 +40,33 @@ public class Config : IConfig
         private const bool REDIRECT_HTTP_DEFAULT = true;
         private const bool STS_ENABLED_DEFAULT = true;
         private const int STS_PERIOD_DEFAULT = 2592000;
-        private const int MAX_PAYLOAD_SIZE_DEFAULT = 131072;
-        private const bool CONFIG_STATUS_DEFAULT = false;
-        private const string LOG_LEVEL_DEFAULT = "Warn";
+        private const int MAX_PAYLOAD_SIZE_DEFAULT = 102400;
```
Comment (jillcary) [inline ProxyAgent/Runtime/Config.cs]: Where does this number come from? A comment could be helpful for when we look back on this later.

File: ProxyAgent/ProxyMiddleware.cs (hunk 1 above)
Comment (dluc) [reply ProxyAgent/ProxyMiddleware.cs]: a debugging "helper" to distinguish one request from another, during development phases

File: ProxyAgent/Runtime/Config.cs (hunk 2 above)
Comment (dluc) [reply ProxyAgent/Runtime/Config.cs]: 👍

**Pull request merged**